src/main.rs

```diff
@@ -175,6 +189,9 @@ impl Results {
     }
 }
 
+// This is not yet designed to handle combinations of different parameters.
+// Combining more than one set of parameters, e.g. versions and ciphers, might
+// result in combinatorial explosion, as intended. But that's not guaranteed.
```

Comment:
I don't think that this comment quite works for what you have here.  What you have here doesn't appear to be combinatorial.

Reply:
I rewrote this comment to express better what I meant to say here.